### PR TITLE
fix: upload gstr1 data on gst-portal in parts

### PR DESCRIPTION
**Pull request by Sanket322**

### CODE REVIEW

File: india_compliance/gst_india/doctype/gstr_1_beta/gstr_1_beta.js [hunk 1]
```diff
@@ -444,15 +444,19 @@ class GSTR1 {
...
     render_form_actions() {
-        if (this.data && (!is_gstr1_api_enabled() || this.status == "Filed")) return;
+        if (this.data && !is_gstr1_api_enabled()) return;
 
         this.gstr1_action = new GSTR1Action(this.frm);
 
         // Custom Buttons
-        if (this.data)
+        if (this.data){
+            if (this.status == "Filed")
```
Comment (ljain112) [inline india_compliance/gst_india/doctype/gstr_1_beta/gstr_1_beta.js]: show button enen ig api is not eneable

File: india_compliance/gst_india/doctype/gstr_1_beta/gstr_1_beta.js [hunk 2]
```diff
@@ -712,6 +716,22 @@ class GSTR1 {
         let element = $('[data-fieldname="data_section"]');
         element.prepend(gst_liability_html);
     }
+
+    get_journal_entries(){
+        const { company, month_or_quarter, year } = this.frm.doc;
+
+        frappe.call({
+            method: "india_compliance.gst_india.doctype.gstr_1_beta.gstr_1_beta.get_general_entries",
+            args: {month_or_quarter, year, company},
```
Comment (ljain112) [inline india_compliance/gst_india/doctype/gstr_1_beta/gstr_1_beta.js]: function naming

File: india_compliance/gst_india/doctype/gstr_1_beta/gstr_1_beta.py [hunk 3]
```diff
@@ -226,6 +226,84 @@ def update_filing_status(filters):
...
+
+    gst_accounts = get_gst_accounts_by_type(company, "Sales Reverse Charge")
+    if not gst_accounts:
+        return "No GST Accounts found"
+
+    return (
+        frappe.qb.from_(journal_entry)
+        .join(journal_entry_account)
+        .on(journal_entry.name == journal_entry_account.parent)
+        .select(
```
Comment (ljain112) [inline india_compliance/gst_india/doctype/gstr_1_beta/gstr_1_beta.py:252]: do not return whole data

File: india_compliance/gst_india/doctype/gstr_1_beta/gstr_1_beta.py [hunk 4]
```diff
@@ -226,6 +226,84 @@ def update_filing_status(filters):
     frappe.db.set_value("GST Return Log", log_name, "filing_status", "Not Filed")
 
 
+@frappe.whitelist()
+def get_general_entries(month_or_quarter, year, company):
```
Comment (ljain112) [inline india_compliance/gst_india/doctype/gstr_1_beta/gstr_1_beta.py]: function name

File: india_compliance/gst_india/doctype/gstr_1_beta/gstr_1_beta.js [hunk 5]
```diff
@@ -2507,6 +2527,33 @@ class GSTR1Action extends FileGSTR1Dialog {
...
+            ],
+            primary_action_label: "Create",
+            primary_action: async (values) => {
+                const { company, company_gstin, month_or_quarter, year } = this.frm.doc;
+
+                frappe.call({
+                    method: "india_compliance.gst_india.doctype.gstr_1_beta.gstr_1_beta.make_journal_entry",
+                    args: { company, company_gstin, month_or_quarter, year, auto_submit: values.auto_submit },
+                    callback: (r) => {
+                        frappe.set_route("journal-entry", r.message);
```
Comment (ljain112) [inline india_compliance/gst_india/doctype/gstr_1_beta/gstr_1_beta.js:2547]: open in new tab

File: india_compliance/gst_india/doctype/gstr_1_beta/gstr_1_beta.py [hunk 6]
```diff
@@ -226,6 +226,84 @@ def update_filing_status(filters):
     frappe.db.set_value("GST Return Log", log_name, "filing_status", "Not Filed")
 
 
+@frappe.whitelist()
```
Comment (ljain112) [inline india_compliance/gst_india/doctype/gstr_1_beta/gstr_1_beta.py:232]: permission check

File: india_compliance/gst_india/doctype/gstr_1_beta/gstr_1_beta.py [hunk 7]
```diff
@@ -226,6 +226,84 @@ def update_filing_status(filters):
...
+            sales_invoice_taxes.account_head.as_("account"),
+            Sum(sales_invoice_taxes.tax_amount).as_("tax_amount"),
+        )
+        .where(sales_invoice.is_reverse_charge == 1)
+        .where(
+            Date(sales_invoice.posting_date).between(
+                getdate(from_date), getdate(to_date)
+            )
+        )
+        .groupby(sales_invoice_taxes.account_head)
```
Comment (ljain112) [inline india_compliance/gst_india/doctype/gstr_1_beta/gstr_1_beta.py:296]: where condition gst_tax_type is not null

File: india_compliance/gst_india/doctype/gstr_1_beta/gstr_1_beta.js [hunk 8]
```diff
@@ -2441,6 +2221,506 @@ class DetailViewDialog {
...
+                .set_description(`<p style="color: red">OTP is either expired or incorrect.</p>`);
+
+            this.toggle_actions(true);
+            return;
+        }
+
+        this.filing_dialog.hide();
+
+        if (response.error?.error_cd === "RET09001") {
+            this.frm.page.set_primary_action("Proceed to File", () =>
```
Comment (vorasmit) [inline india_compliance/gst_india/doctype/gstr_1_beta/gstr_1_beta.js]: Upload Button.

**Pull request merged**